packages/core/src/resolver.js

```diff
@@ -19,15 +18,18 @@ export default class Resolver {
 
     // ESM Loader
     this.esm = esm(module)
+
+    // module resolver
+    this.moduleResolver = moduleResolver || require.resolve
   }
 
   resolveModule(path) {
     try {
-      return Module._resolveFilename(path, {
+      return this.moduleResolver(path, {
         paths: this.options.modulesDir
       })
     } catch (error) {
-      if (error.code === 'MODULE_NOT_FOUND') {
+      if (error.code === 'MODULE_NOT_FOUND' || error.message.includes('Cannot resolve module')) {
```

Comment:
@Globegitter Could you please add a link here https://github.com/nuxt/nuxt.js/pull/5796

@pi0 What do you think adding `process.env.NODE_ENV === 'test'` ?

Reply:
Probably we can open a repo in jest for adding correct code and reference it here.

Reply:
https://github.com/facebook/jest/pull/8487 :+1:  :D

Reply:
@pi0 See my comment, I've opened pr in jest, but I suppose that they may not approve and release in short term.